### PR TITLE
ClearlyDefined harvest API fixes

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: scanner/src/funTest/kotlin/storages/ClearlyDefinedStorageFunTest.kt [hunk 1]
```diff
@@ -121,4 +122,110 @@ class ClearlyDefinedStorageFunTest : StringSpec({
             )
         }
     }
+
+    "Scan results for packages without a namespace should be read correctly" {
+        val id = Identifier("NPM::iobroker.eusec:0.9.9")
+        val result = storage.read(id)
+
+        result.shouldBeSuccess {
+            it shouldContain ScanResult(
```
Comment (fviernau) [inline scanner/src/funTest/kotlin/storages/ClearlyDefinedStorageFunTest.kt:131]: There is already a test which covers a complete scan result above.
This assertion is quite detailed. I wonder if it is asserting "too much". Maybe just something like:
```
result.shouldBeSuccess {
    it.find { it.provenance.resolvedRevision == "327b125548c9b806490085a2dacfdfc6e7776803" } shouldNot beNull()
}
```

**Review by fviernau — CHANGES_REQUESTED** [review]

File: clients/clearly-defined/src/funTest/kotlin/ClearlyDefinedServiceFunTest.kt [hunk 2]
```diff
@@ -68,7 +68,7 @@ class ClearlyDefinedServiceFunTest : WordSpec({
             val curation = service.getCuration(
                 coordinates.type,
                 coordinates.provider,
-                coordinates.namespace.orEmpty(),
+                coordinates.namespace ?: "-",
```
Comment (fviernau) [inline clients/clearly-defined/src/funTest/kotlin/ClearlyDefinedServiceFunTest.kt:71]: nit: Should the commit message body emphasize that this fixes broken requests?

File: scanner/src/funTest/kotlin/storages/ClearlyDefinedStorageFunTest.kt (hunk 1 above)
Comment (sschuberth) [reply scanner/src/funTest/kotlin/storages/ClearlyDefinedStorageFunTest.kt]: Done in a similar but slight more verbose way.

**Review by fviernau — APPROVED** [review]

**Comment by sschuberth** [comment]
Merging despite the unrelated SW360 test failures, see https://www.eclipsestatus.io/.

**Pull request merged**